vast/vast.cpp

```diff
@@ -50,9 +80,51 @@ int main(int argc, char** argv) {
     return EXIT_FAILURE;
   }
   // Application setup.
-  const auto [root, factory] = make_application(argv[0]);
+  auto [root, root_factory] = make_application(argv[0]);
   if (!root)
     return EXIT_FAILURE;
+  // Load plugins.
+  auto& plugins = plugins::get();
+  auto plugin_dirs = detail::get_plugin_dirs(cfg);
+  for (const auto& dir : plugin_dirs) {
```

Comment:
I think it would be good to require a more explicit opt-in for plugin loading, this is executing arbitrary code after all and everyone with access to the current users home directory can inject it.

Not completely sure what the best solution would be, but some ideas:
 - a build-time switch to turn off plugin support completely
 - a run-time --enable-plugins switch
 - a list of specific dynamic plugins to load, intead of directories, like `--enable-plugin=/usr/lib/vast/plugins/matcher.so`

Reply:
We'll postpone this to a separate story so we can discuss this prperly.

Reply:
Can we at least turn it off by default then until it is discussed?